cmd/piper.go

```diff
@@ -162,6 +164,65 @@ func PrepareConfig(cmd *cobra.Command, metadata *config.StepData, stepName strin
 	return nil
 }
 
+func convertTypes(config map[string]interface{}, options interface{}) map[string]interface{} {
+	typedOptions := reflect.ValueOf(options)
+	if typedOptions.Kind() == reflect.Ptr {
+		typedOptions = typedOptions.Elem()
+	}
+	optionsType := typedOptions.Type()
+
+	for key := range config {
+		// Find the field in options which will be mapped to 'key'
+		field := findField(key, typedOptions, optionsType)
+		if field == nil {
+			continue
+		}
+
+		entry := reflect.ValueOf(config[key])
```

Comment:
Naming: entry  vs field. What is what? Where is the expected type saved and where the actual.

Reply:
The expected type is whatever concrete type is declared for the step options struct's field. The resolved config just maps from string to interface{}. We need to find out the actual type of both via reflection. Hopefully I've made it more clear by using "paramName" and "paramValue" and "paramValueType". Do you see a better way?